src/module/rules/rule-element/damage-dice.ts

```diff
@@ -54,41 +54,53 @@ class DamageDiceRuleElement extends RuleElementPF2e<DamageDiceRuleSchema> {
 
         const deferredDice = (params: DeferredValueParams = {}): DamageDicePF2e | null => {
             const label = this.getReducedLabel();
-            const diceNumber = Number(this.resolveValue(this.diceNumber, 0, { resolvables: params.resolvables })) || 0;
 
-            const resolvedBrackets = this.resolveValue(this.brackets, {}, { resolvables: params.resolvables });
+            // If this rule element's predicate would have passed without all fields being resolvable, send out a
+            // warning.
+            const testPassed =
+                this.predicate.length === 0
+                    ? true
+                    : this.resolveInjectedProperties(this.predicate).test(
+                          params.test ?? this.actor.getRollOptions(["damage"])
+                      );
```

Comment:
Can't do a suggestion cause no prettier here to fix it for me, but 
```
const testPassed = !this.predicate.length || this.resolveInjectedProperties(this.predicate).test(
    params.test ?? this.actor.getRollOptions(["damage"])
);```

Reply:
That is less readable, but ok